__tests__/cookie-functions.test.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * @jest-environment jsdom
+ */
+
+/* eslint-env jest */
+
+import * as CookieHelpers from '../src/javascripts/cookie-functions'
+
+describe('Cookie settings', () => {
+  describe('Reading a cookie', () => {
+    beforeEach(() => {
+      // Allow setting _ga cookie
+      CookieHelpers.setConsentCookie({ analytics: true })
+    })
+
+    afterEach(() => {
+      // Delete test cookies
+      document.cookie = '_ga=;expires=Thu, 01 Jan 1970 00:00:00 UTC'
+      document.cookie = 'design_system_cookies_policy=;expires=Thu, 01 Jan 1970 00:00:00 UTC'
+    })
+
+    it('returns null if no cookie present', async () => {
+      expect(CookieHelpers.Cookie('_ga')).toEqual(null)
+    })
+
+    it('returns cookie if present', async () => {
+      CookieHelpers.Cookie('_ga', 'foo')
+      expect(CookieHelpers.Cookie('_ga')).toEqual('foo')
+    })
+  })
+
+  describe('Setting a cookie', () => {
+    afterEach(() => {
+      // Delete test cookies
+      document.cookie = 'myCookie=;expires=Thu, 01 Jan 1970 00:00:00 UTC'
+      document.cookie = 'design_system_cookies_policy=;expires=Thu, 01 Jan 1970 00:00:00 UTC'
+    })
+
+    it('doesnt set a cookie with a value if not a recognised name', async () => {
+      CookieHelpers.Cookie('myCookie', 'myValue')
+
+      expect(document.cookie).toEqual('')
+    })
+
+    it('allows deletion of any cookie even if not recognised', async () => {
+      document.cookie = 'myCookie=hello; path=/'
+      document.cookie = 'otherCookie=world; path=/'
+
+      CookieHelpers.Cookie('otherCookie', null)
+
+      expect(document.cookie).toEqual('myCookie=hello')
+    })
+
+    it('sets allowed cookie with no options', async () => {
+      CookieHelpers.Cookie('design_system_cookies_policy', '{"analytics":false}')
+
+      expect(document.cookie).toEqual('design_system_cookies_policy={"analytics":false}')
+    })
+
+    it('sets allowed cookie with options', async () => {
+      CookieHelpers.Cookie('design_system_cookies_policy', '{"analytics":false}', { days: 100 })
+
+      // Annoyingly JS can't retrieve expiry date directly from document.cookie, this is all we can assert
+      expect(document.cookie).toEqual('design_system_cookies_policy={"analytics":false}')
+    })
+  })
+
+  describe('getConsentCookie', () => {
+    afterEach(() => {
+      // Delete consent cookie
+      document.cookie = 'design_system_cookies_policy=;expires=Thu, 01 Jan 1970 00:00:00 UTC'
+    })
+
+    it('returns null if consent cookie not present', async () => {
+      expect(CookieHelpers.getConsentCookie()).toEqual(null)
+    })
+
+    it('returns consent cookie object if present', async () => {
+      document.cookie = 'design_system_cookies_policy={"analytics":false}'
+
+      expect(CookieHelpers.getConsentCookie()).toEqual({ analytics: false })
+    })
+  })
+
+  describe('setConsentCookie', () => {
```

Comment:
Do we need something which tests that a consent cookie is set with the default values if no values are passed in?

Reply:
Good idea

Reply:
Done